ui/components/NotificationCenter/constants.js

```diff
@@ -37,4 +39,86 @@ export const SEVERITY_STYLE = {
     color : NOTIFICATIONCOLORS.WARNING
   },
 
+}
+
+//TODO: This should be generated from OPENAPI schema
+const EVENT_SCHEMA = {
+  type : "object",
+  properties : {
+    id : { type : "string" },
+    description : {
+      type : "string",
+      default : ""
+    },
+    severity : {
+      type : "string",
+      enum : Object.values(SEVERITY),
+      default : SEVERITY.INFO
+    },
+    status : {
+      type : "string",
+      enum : Object.values(STATUS),
+      default : STATUS.UNREAD
+    },
+    created_at : { type : "string" },
+    updated_at : { type : "string" },
+    user_id : { type : "string" },
+    system_id : { type : "string" },
+    operation_id : { type : "string" },
+    action : { type : "string" },
+    metadata : {
+      type : "object",
+    }
+  },
+  required : ["id", "severity", "status", "created_at", "updated_at", "user_id", "system_id", "action"]
+}
+
+
+// Validate event against EVENT_SCHEMA and return [isValid,validatedEvent]
+export const validateEvent = (event) => {
+  const eventCopy = _.cloneDeep(event) || {};
+  const ajv = new Ajv({
+    useDefaults : true,
+
+  });
+  const validate = ajv.compile(EVENT_SCHEMA);
+  const valid = validate(eventCopy);
+  return [valid, eventCopy];
+}
+
+// return validated events (adds default values if not present)
+export const validateEvents = (events) => {
+  return events.map((event) => {
+    const [isValid, validatedEvent] = validateEvent(event)
+    return isValid ? validatedEvent : null
+  }).filter((event) => event)
+}
+
+
+
+const EVENT_METADATA_SCHEMA = {
```

Comment:
This isn’t always the case, in long operations i.e. an operation which invoked multiple other operation the metadata contains a summary which is not an error specifically.

Reply:
Thanks Uzair , i have not considered that . do you have an  example response for this kind of situation